### ISSUE TITLE
Split out Text Checkbox Group Field

### ISSUE THREAD

**Issue by mak001** [dnadesign/silverstripe-elemental#227]
Can `DNADesign\Elemental\Forms\TextCheckboxGroupField` be split out into its own module?
I can see this being useful in some other modules that would not need elemental.

**Comment by muskie9** [comment]
@robbieaverill I can start this if you want to create a repository with an initial commit. We have a project we'd pull in the new module that we're wrapping up an initial build for so might be good timing.

**Comment by robbieaverill** [comment]
cc @brynwhyman - the React elemental rewrite would be a good opportunity to get this done

**Comment by muskie9** [comment]
Sorry for the radio silence, I plan on getting back to this over the next week.

**Comment by muskie9** [comment]
Just curious if anyone else has experience the field laying out a bit odd currently. I think it's in v2, but seems to be fine in v3. I'll try to get a screenshot of this, but wasn't sure if anything changed between those versions for that field.

Additionally, what branch should this PR go against. I have it about 2/3 done at this point and have a long weekend coming up so hoping to get it to a state to open the PR.

**Comment by robbieaverill** [comment]
@muskie9 yeah there is a slightly different set of CSS classes between Bootstrap 4.0.0 beta and stable (which changed between SilverStripe 4.0/4.1 and 4.2 I think). That could be it. Related issue: https://github.com/dnadesign/silverstripe-elemental/issues/201

>what branch should this PR go against

In this repository, master since it's a breaking change.

I'm discussing internally about what the repository that the field goes into should be called.

Thanks for doing this!

**Comment by robbieaverill** [comment]
@muskie9 I've created https://github.com/silverstripe/silverstripe-checkboxgroupfield as a new repository for this change

**Comment by muskie9** [comment]
Thanks @robbieaverill, would you be able to do a quick README, I'm unable to fork it at this time with no files present.

**Comment by robbieaverill** [comment]
Sorry @muskie9, I've added an empty readme

**Comment by muskie9-opensource** [comment]
@robbieaverill I deeply apologize for the drop-off with this ticket and subsequently the repository. I do think this is still something valuable in the community as we have all become accustomed to a "value and/or show" type of structure with elemental.

I'd love to pick this back up, however I want to check if https://github.com/silverstripe/silverstripe-checkboxgroupfield is still a valid place for that. I recall digging in a bit and the expectation of field to save is a bit of work which is fine, I just want to be sure we're setting this where it's most accessible to the community as it is completed and is stable.

Again, sorry for the complete drop-off with this ticket, I'm glad it didn't get closed due to it's staleness.
